python/src/lib.rs

```diff
@@ -362,12 +472,15 @@ impl RawDeltaTable {
         post_commithook_properties: Option<PyPostCommitHookProperties>,
     ) -> PyResult<Vec<String>> {
         let (table, metrics) = py.allow_threads(|| {
-            let mut cmd = VacuumBuilder::new(
-                self._table.log_store(),
-                self._table.snapshot().map_err(PythonError::from)?.clone(),
-            )
-            .with_enforce_retention_duration(enforce_retention_duration)
-            .with_dry_run(dry_run);
+            let snapshot = match self._table.lock() {
+                Ok(table) => Ok(table.snapshot().cloned().map_err(PythonError::from)?),
+                Err(e) => Err(e),
+            }
+            .expect("SHOOT!");
+            // TODO: XXX
```

Comment:
artifact?

Reply:
d'oh! good catch, I did miss some of this junk in my diff review before poooshing